doc/source/make_tables.py

```diff
@@ -432,23 +434,170 @@ def get_row(cls, i, row_data):
         return cls.row_template.format(name, row_data['hex'], row_data['hex'])
 
 
-def _sort_colors():
-    colors_rgb = [pv.Color(c).float_rgb for c in pv.hexcolors.values()]
-    # Sort colors by hue, saturation, and value (HSV)
-    return sorted(colors_rgb, key=lambda c: tuple(colorsys.rgb_to_hsv(*pv.Color(c).float_rgb)))
+def _sort_colors_by_hls(colors: Sequence[Color]):
+    return sorted(colors, key=lambda c: c._float_hls)
+
+
+ALL_COLORS: tuple[Color] = tuple(pv.Color(c) for c in pv.hexcolors.keys())
+
+# Saturation constants
+GRAYS_SATURATION_THRESHOLD = 0.15
+
+# Lightness constants
+LOWER_LIGHTNESS_THRESHOLD = 0.1
+UPPER_LIGHTNESS_THRESHOLD = 0.9
+
+# Hue constants in range [0, 1]
+_360 = 360.0
+RED_UPPER_BOUND = 10 / _360
+ORANGE_UPPER_BOUND = 39 / _360
+YELLOW_UPPER_BOUND = 61 / _360
+GREEN_UPPER_BOUND = 157 / _360
+CYAN_UPPER_BOUND = 187 / _360
+BLUE_UPPER_BOUND = 248 / _360
+VIOLET_UPPER_BOUND = 290 / _360
+MAGENTA_UPPER_BOUND = 348 / _360
+
+
+class ColorClassification(StrEnum):
+    WHITE = auto()
+    BLACK = auto()
+    GRAY = auto()
+    RED = auto()
+    YELLOW = auto()
+    ORANGE = auto()
+    GREEN = auto()
+    CYAN = auto()
+    BLUE = auto()
+    VIOLET = auto()
+    MAGENTA = auto()
+
+
+def classify_color(color: Color) -> ColorClassification:
```

Comment:
As a follow-up PR, this could be integrated directly with the `Color` class if desired, e.g. as a `Color.classification` property. Though, the classifications aren't standard in any way, the thresholds used for the classification were customized to make the tables in this PR make sense and have a decent spread across different categories.